hudi-common/src/main/java/org/apache/hudi/metadata/BaseTableMetadata.java

```diff
@@ -157,8 +157,13 @@ protected BaseTableMetadata(HoodieEngineContext engineContext, HoodieMetadataCon
     List<String> partitions = Collections.emptyList();
     if (hoodieRecord.isPresent()) {
       if (!hoodieRecord.get().getData().getDeletions().isEmpty()) {
```

Comment:
this code section is being repeated. Maybe we can refactor into a method like:

  checkMetadataEntry(hoodieRecord.get().getData())

Reply:
+1